### ISSUE TITLE
[M10][infra] Three-class storage layout (poc-distill / shared / reserve-pretrain)

### ISSUE THREAD

**Issue by travisgalloway** [travisgalloway/monica#97]
Part of #65. Extends the R2 layout in #80.

**Phase 1 — storage layout.** Adopt the **three-class** artifact layout that drives the storage
plan and keeps the student layout downstream of every frozen artifact.

### The three classes
- **POC-distill-specific** (`poc-distill/`) — the tokenized distillation corpus (#C) and teacher
  outputs (#E). Drives the distillation matching only.
- **Shared** (`shared/`) — the instruct (#F), reasoning-trace (#G), and tool-use (#M) SFT corpora
  and the verifiable RL sets (#N). Both the POC and the production model post-train the same way,
  so these are curated once and reused everywhere.
- **Production-reserve** (`reserve-pretrain/`) — the full from-scratch corpus (#70/#71), built
  only after a layout validates.

```
poc-distill/      corpus/{cleaned,tokenized/qwen25-8k}/  teacher-outputs/{topk-logits,hidden-states}/  manifests/
shared/           sft/{cleaned,tokenized/qwen25-8k}/  rl/{math-verifiable,code-verifiable}/  eval/
reserve-pretrain/ cleaned/  tokenized/v1-qwen25-8k/  manifests/
```

### Rules
- Keep cleaned text and RL problems **tokenizer-agnostic and durable**; pin tokenizer + seq_len
  in the name of every tokenized folder.
- Store a **long** packing; mark document boundaries; pack reasoning traces atomically.

### Acceptance
- [ ] R2 prefixes created per the layout; readers/writers (#80) point at the right prefixes.
- [ ] Tokenized folders name-pin tokenizer + seq_len; cleaned/RL stay tokenizer-free.

Labels: data.

**Comment by travisgalloway** [comment]
PR #126 adds `src/data/storage.py` as the three-class layout authority (poc-distill / shared / reserve-pretrain) with tokenizer+seq_len name-pinning and tokenizer-free cleaned/RL prefixes; the #92/#95/#96 drivers now route through it (paths unchanged). Full suite green (399 passed).
